### ISSUE TITLE
Registration in Windows Task Scheduler with Cron expressions

### ISSUE THREAD

**Issue by SebastiaanPolfliet** [dahall/TaskScheduler#784]
Hi,

When testing the library with cron expressions, I noticed a subtle difference in the way the task gets registered in the Windows Task Scheduler. Without cron expressions I get the following : 

![image](https://user-images.githubusercontent.com/6726678/46079559-bd014680-c197-11e8-9087-4b2a22b981c7.png)

If I use a cron expression to express the same (like 15 4 * * 3), the task gets registered as: 

![image](https://user-images.githubusercontent.com/6726678/46079614-ee7a1200-c197-11e8-991d-aa0a4b394e4c.png)

You see that both the Months and Weeks (by First/Second/Third/Fourth/Last) are written out in expanded form.

I know there is no functional difference, but my end-users are really keen on the way it is presented.

If you need more information (code samples) to reproduce the case, do not hesitate to contact me. 

Sebastiaan

**Comment by dahall** [comment]
My guess is that there is a subtle difference in the actual triggers. Will you dump both those instances referenced above to XML and post them here? It will help me determine what I can do to the cron parser or suggest you do to resolve the issue.

**Comment by SebastiaanPolfliet** [comment]
Thank you for your quick reply. I think the issue is related to the fact that a weekly CronExpression gets translated to a `MonthlyDOWTrigger` and before I used `WeeklyTrigger`.

You can make the same observation in the XML files (I've extracted the triggers):

Manual (WeeklyTrigger) 
```
<Triggers>
    <CalendarTrigger>
      <StartBoundary>2018-09-26T05:00:00</StartBoundary>
      <Enabled>true</Enabled>
      <ScheduleByWeek>
        <DaysOfWeek>
          <Wednesday />
        </DaysOfWeek>
        <WeeksInterval>1</WeeksInterval>
      </ScheduleByWeek>
    </CalendarTrigger>
  </Triggers>
```

Cron trigger ("0 5 * * 3") 
```
  <Triggers>
    <CalendarTrigger>
      <StartBoundary>2018-09-26T05:00:00</StartBoundary>
      <Enabled>true</Enabled>
      <ScheduleByMonthDayOfWeek>
        <Weeks>
          <Week>1</Week>
          <Week>2</Week>
          <Week>3</Week>
          <Week>4</Week>
          <Week>Last</Week>
        </Weeks>
        <DaysOfWeek>
          <Wednesday />
        </DaysOfWeek>
        <Months>
          <January />
          <February />
          <March />
          <April />
          <May />
          <June />
          <July />
          <August />
          <September />
          <October />
          <November />
          <December />
        </Months>
      </ScheduleByMonthDayOfWeek>
    </CalendarTrigger>
  </Triggers>
```

**Comment by dahall** [comment]
So the cron expression generates the WeeklyTrigger or the MonthlyDOWTrigger? If it generates the WeeklyTrigger, then I'm inclined to leave it as that is more efficient. If the other, then I will add a condition in the code to look for that instance and use the WeeklyTrigger.

**Comment by SebastiaanPolfliet** [comment]
Sorry, I made a mistake when pasting the xmls. I've updated my post. 

To clear up the confusion: The Cron expression generates a MonthlyDOWTrigger.

**Comment by SebastiaanPolfliet** [comment]
Do you need more information?

**Comment by dahall** [comment]
Fixed in 2.8.6, available now.

**Comment by dahall** [comment]
Please test and verify. I ran about 40 patterns through it, but I'm sure I missed some cases. There shouldn't be any more limitations, i.e. it should support all cron patterns now.
